File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        memo = [1] * len(nums)
        n = len(nums) - 1
        
        for i in range(n, -1, - 1):
            for j in range(i, n + 1):
                if nums[j] > nums[i]: 
                    memo[i] = max(memo[i], 1 + memo[j])
                    
        return max(memo)

File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let max = nums[0];
        const map = {};

        for (const num of nums) {
            if (num in map) {
                map[num]++;
            } else {
                map[num] = 1;
            };

            if (map[num] > map[max]) {
                max = num;
            };
        };

        return parseInt(max);
    };
};
